packages/components/src/calendar-react-day-picker/date-calendar.stories.tsx

```diff
@@ -0,0 +1,327 @@
+import { fn } from '@storybook/test';
+import { fr, ja, es, ko, pt, ar, it } from 'date-fns/locale';
+import { useState } from 'react';
+import { DateCalendar } from './index';
+import type { Meta, StoryObj } from '@storybook/react';
+
+const meta: Meta< typeof DateCalendar > = {
+	title: 'Components/DateCalendar',
+	component: DateCalendar,
+	tags: [ 'autodocs' ],
+	parameters: {
+		controls: { expanded: true },
+	},
+	argTypes: {
+		locale: {
+			options: [ 'ar', 'es', 'fr', 'ja', 'ko', 'pt', 'it' ],
+			mapping: {
+				ar: ar,
+				es: es,
+				fr: fr,
+				ja: ja,
+				ko: ko,
+				pt: pt,
+				it: it,
+			},
+			control: {
+				type: 'select',
+			},
+		},
+		labels: {
+			control: false,
+		},
+		defaultSelected: { control: { type: 'date' } },
+		selected: { control: { type: 'date' } },
+		onSelect: {
+			control: false,
+		},
+		defaultMonth: { control: { type: 'date' } },
+		month: { control: { type: 'date' } },
+		onMonthChange: {
+			control: false,
+		},
+		endMonth: { control: { type: 'date' } },
+		startMonth: { control: { type: 'date' } },
+		footer: { control: { type: 'text' } },
+	},
+	args: {
+		onMonthChange: fn(),
+		onSelect: fn(),
+	},
+};
+export default meta;
+
+type Story = StoryObj< typeof DateCalendar >;
+
+export const Default: Story = {};
+
+export const TimeZone: Story = {
+	args: {
+		timeZone: 'Asia/Tokyo',
+		// In 1 week, but in Tokyo timezone
+		defaultSelected: new Date( new Date().setDate( new Date().getDate() + 7 ) ),
+		disabled: [
+			{
+				// Disable any date before today, but in Tokyo timezone
+				before: new Date(),
+			},
+		],
+	},
+};
+
+export const Controlled: Story = {
+	render: function ControlledDateCalendar( args ) {
+		const [ selected, setSelected ] = useState< Date >();
+		return (
+			<DateCalendar
+				{ ...args }
+				selected={ selected }
+				onSelect={ ( selectedDate, ...rest ) => {
+					setSelected( selectedDate );
+					// TS is strict about `onSelect` expecting a non-undefined date
+					// when the selection is required.
+					if ( ! args.required ) {
+						args.onSelect?.( selectedDate, ...rest );
+					} else if ( selectedDate ) {
+						args.onSelect?.( selectedDate, ...rest );
+					}
```

Comment:
Can't we combine in one condition though?

```suggestion
					// TS is strict about `onSelect` expecting a non-undefined date
					// when the selection is required.
					if ( ! args.required || selectedDate ) {
						args.onSelect?.( selectedDate, ...rest );
					}
```

Reply:
That won't work, because `args.onSelect` is of type `OnSelectHandler<Date> | OnSelectHandler<Date | undefined>`.

To disambiguate between the two types that make the union, the only working way I found was to introduce an implicit typeguard on the `required` prop. Alternatively, we could add a `@ts-expect-error` comment.

Do you have other ideas?

Reply:
Got it. We're good then, no need to alter.